op-node/chaincfg/chains.go

```diff
@@ -2,135 +2,100 @@ package chaincfg
-	L1SystemConfigAddress:  common.HexToAddress("0xAe851f927Ee40dE99aaBb7461C00f9622ab91d60"),
-	RegolithTime:           u64Ptr(1679079600),
-}
-
-var Sepolia = rollup.Config{
-	Genesis: rollup.Genesis{
-		L1: eth.BlockID{
-			Hash:   common.HexToHash("0x48f520cf4ddaf34c8336e6e490632ea3cf1e5e93b0b2bc6e917557e31845371b"),
-			Number: 4071408,
-		},
-		L2: eth.BlockID{
-			Hash:   common.HexToHash("0x102de6ffb001480cc9b8b548fd05c34cd4f46ae4aa91759393db90ea0409887d"),
-			Number: 0,
-		},
-		L2Time: 1691802540,
-		SystemConfig: eth.SystemConfig{
-			BatcherAddr: common.HexToAddress("0x8F23BB38F531600e5d8FDDaAEC41F13FaB46E98c"),
-			Overhead:    eth.Bytes32(common.HexToHash("0x00000000000000000000000000000000000000000000000000000000000000bc")),
-			Scalar:      eth.Bytes32(common.HexToHash("0x00000000000000000000000000000000000000000000000000000000000a6fe0")),
-			GasLimit:    30000000,
-		},
-	},
-	BlockTime:              2,
-	MaxSequencerDrift:      600,
-	SeqWindowSize:          3600,
-	ChannelTimeout:         300,
-	L1ChainID:              big.NewInt(11155111),
-	L2ChainID:              big.NewInt(11155420),
-	BatchInboxAddress:      common.HexToAddress("0xff00000000000000000000000000000011155420"),
-	DepositContractAddress: common.HexToAddress("0x16fc5058f25648194471939df75cf27a2fdc48bc"),
-	L1SystemConfigAddress:  common.HexToAddress("0x034edd2a225f7f429a63e0f1d2084b9e0a93b538"),
-	RegolithTime:           u64Ptr(0),
-}
+var Mainnet, Goerli, Sepolia *rollup.Config
 
-var NetworksByName = map[string]rollup.Config{
-	"goerli":  Goerli,
-	"mainnet": Mainnet,
-	"sepolia": Sepolia,
+func init() {
+	mustCfg := func(name string) *rollup.Config {
+		cfg, err := GetRollupConfig(name)
+		if err != nil {
+			panic(fmt.Errorf("failed to load rollup config %q: %w", name, err))
+		}
+		return cfg
+	}
+	Mainnet = mustCfg("op-mainnet")
+	Goerli = mustCfg("op-goerli")
+	Sepolia = mustCfg("op-sepolia")
 }
 
-var L2ChainIDToNetworkName = func() map[string]string {
+var L2ChainIDToNetworkDisplayName = func() map[string]string {
 	out := make(map[string]string)
-	for name, netCfg := range NetworksByName {
-		out[netCfg.L2ChainID.String()] = name
+	for _, netCfg := range superchain.OPChains {
+		out[fmt.Sprintf("%d", netCfg.ChainID)] = netCfg.Name
 	}
 	return out
 }()
 
+// AvailableNetworks returns the selection of network configurations that is available by default.
+// Other configurations that are part of the superchain-registry can be used with the --beta.network flag.
 func AvailableNetworks() []string {
+	return []string{"op-mainnet", "op-goerli", "op-sepolia"}
+}
+
+// BetaAvailableNetworks returns all available network configurations in the superchain-registry.
+// This set of configurations is experimental, and may change at any time.
+func BetaAvailableNetworks() []string {
 	var networks []string
-	for name := range NetworksByName {
-		networks = append(networks, name)
+	for _, cfg := range superchain.OPChains {
```

Comment:
Is `OPChains` a strict superset of `{op-mainnet, op-goerli, op-sepolia}`?

Reply:
Yes, `superchain.OPChains` contains all chains that are in the superchain-registry, which includes some others than the OP chains.